sdks/python/scripts/run_pytest.sh

```diff
@@ -28,10 +28,10 @@ envname=${1?First argument required: suite base name}
 posargs=$2
 
 # Run with pytest-xdist and without.
-python setup.py pytest --addopts="-o junit_suite_name=${envname} \
+pytest --addopts="-o junit_suite_name=${envname} \
```

Comment:
You should be able to remove the `--addopts` flag and leave the value part.

Reply:
Ah yes, I was wrangling with that the other day.  It would simplify things if we could just get rid of run_pytest.sh.  Do we still need to run the tests with and without xdist?

Reply:
Yes, there are a few tests that are still incompatible with xdist, such as those using save_main_session.

Reply:
If the goal is to run both tests regardless of whether there is a failure (i.e. avoid early abort) then you can set the tox setting `ignore_errors=true`: https://tox.readthedocs.io/en/latest/config.html#conf-ignore_errors

> If false, a non-zero exit code from one command will abort execution of commands for that environment. If true, a non-zero exit code from one command will be ignored and further commands will be executed. The overall status will be “commands failed”, i.e. tox will exit non-zero in case any command failed.

I was trying to get run_pytest.sh to work with `tox -e py37-pytest -- path/to/test` and I was struggling to get the arg passing correct, and ultimately I just moved it to tox.ini and it worked.   With `ignore_errors=true` in place I don't think the shell script will be necessary.

Reply:
I could try helping with arg passing.

If you specify a certain test to run, one of the pytest runs will certainly fail since it either has or doesn't have the `no_xdist` marker. The script special-cases the 5 exit status from pytest to handle this scenario.

I'm probably missing something, because the way you set things up `tox -e py37-pytest -- path/to/test` should have failed (either the `-m not no_xdist` or the `-m no_xdist` command should have failed).

In any case, I'm strongly in favor of simplifying the build process by having as few shell scripts as possible.

There's also this plugin to suppress certain exit statuses:
https://github.com/yashtodi94/pytest-custom_exit_code